hw/ip/padctrl/data/padctrl_fpv_testplan.hjson

```diff
@@ -0,0 +1,167 @@
+      milestone: V2
+      tests: ["padctrl_assert"]
+    }
+    {
+      name: MioBackwardCheck_A
+      desc: '''If the output mio_attr_o has changed, then in the previous clock cycle, there must be a
+            valid write to the mio_pads register or the write enable signal raises.'''
+      milestone: V2
+      tests: ["padctrl_assert"]
+    }
+    {
+      name: DioWarl_A
+      desc: '''In either generic or Xilinx mode, if a dedic'ed IO's pad attribute is written through
+            the TLUL interface via the dio_pads registers, this assertion checks if the
+            corresponding dio_attr_o's reserved bits remain 0. This assertion intends to test the
+            right_any_read_legal access policy for these dio_pads registers.'''
+      milestone: V2
+      tests: ["padctrl_assert"]
+    }
+    {
+      name: DioAttr_A
+      desc: '''In either generic or Xilinx mode, if a dedicated IO's pad attribute is written through
+            the TLUL interface via dio_pads registers, this assertion checks if the corresponding
+            dio_attr_o value is updated correctly.'''
+      milestone: V2
+      tests: ["padctrl_assert"]
+    }
+    {
+      name: DioBackwardCheck_A
+      desc: '''If the output dio_attr_o has changed, then in the previous clock cycle, there must be a
+            valid write to the dio_pads register or the write enable signal raises.'''
+      milestone: V2
+      tests: ["padctrl_assert"]
+    }
+    {
+      name: Clk_A
+      desc: "Check that clk_pad_i is correctly connected to clk_o."
+      milestone: V2
+      tests: ["padring_assert"]
+    }
+    {
+      name: Rstpad_A
+      desc: "Check that rst_pad_ni is correctly connected to rst_no."
+      milestone: V2
+      tests: ["padring_assert"]
+    }
+    {
+      name: MioIn_A
+      desc: '''Check the muxed IO's output mio_in_o based on mio_pad_io and mio_attr_i inversion
+            bit.'''
+      milestone: V2
+      tests: ["padring_assert"]
+    }
+    {
+      name: MioInBackwardCheck_A
+      desc: '''If muxed IO's output mio_in_o has changed, then mio_pad_io or mio_attr_i must be
+            changed in the same clock cycle.'''
+      milestone: V2
+      tests: ["padring_assert"]
+    }
+    {
+      name: MioOutNormal_A
+      desc: '''If the selected muxed IO is enabled and the corresponding attribute is not open
+            drain, then the mio_pad_io should be the calculated value.'''
+      milestone: V2
+      tests: ["padring_assert"]
+    }
+    {
+      name: MioOutOd0_A
+      desc: '''If the selected muxed IO is enabled and the open drain attribute is set, if the
+            expected mio_output value is 0, then the mio_pad_io should be 0.'''
+      milestone: V2
+      tests: ["padring_assert"]
+    }
+    {
+      name: MioOutOd1_A
+      desc: '''If the selected muxed IO is enabled and the open drain attribute is set, if the
+            expected mio_output value is not 0, then the mio_pad_io could be either 0, 1, x or
```

Comment:
could -> should or must

side comment: I know we have not been very strict about this wording in other FPV testplans, but I feel like we should use "must" instead of "should", since we check for a value and if that value is not correct, the check fails. WDYT?